cpp/src/io/shp/polygon_shapefile_reader.cpp

```diff
@@ -62,12 +83,13 @@ cudf::size_type read_polygon(OGRPolygon const& polygon,
 cudf::size_type read_geometry_feature(OGRGeometry const* geometry,
                                       std::vector<int>& ring_lengths,
                                       std::vector<double>& xs,
-                                      std::vector<double>& ys)
+                                      std::vector<double>& ys,
+                                      const cuspatial::winding_order ring_order)
 {
   OGRwkbGeometryType geometry_type = wkbFlatten(geometry->getGeometryType());
 
   if (geometry_type == wkbPolygon) {
-    return read_polygon(*((OGRPolygon*)geometry), ring_lengths, xs, ys);
+    return read_polygon(*((OGRPolygon*)geometry), ring_lengths, xs, ys, ring_order);
   }
 
   if (geometry_type == wkbMultiPolygon || geometry_type == wkbGeometryCollection) {
```

Comment:
I can't comment on line 96 directly, but it uses a C-style cast. That has a really bad code smell and should be eliminated.  For this code to be correct, you need a `dynamic_cast`.  That may cause this code to not work if compiled with RTTI disabled (we don't do that by default, so I think it's OK).  With `dynamic_cast` you need to check the value of the result for `nullptr` and throw an exception if it is.

Reply:
:+1: